tests/test_cli_socket_autodiscovery.py

```
@@ -63,35 +66,54 @@ def _run(self) -> None:
-                    if b"ping" in data:
-                        conn.sendall(self.response)
-                        return
+            deadline = time.monotonic() + self.accept_timeout
+            while not self._done.is_set():
+                remaining = deadline - time.monotonic()
+                if remaining <= 0:
+                    return
+                server.settimeout(min(0.2, remaining))
+                try:
+                    conn, _ = server.accept()
+                except socket.timeout:
```

Ruff (0.15.13)

[warning] Replace aliased errors with `TimeoutError`

Replace `socket.timeout` with builtin `TimeoutError`

(UP041)